src/frontend/src/lib/types/qr-code.ts

```diff
@@ -0,0 +1,8 @@
+export type QrResult = 'success' | 'cancelled' | 'token_incompatible';
+
+export type QrResponse = {
+	result: QrResult;
```

Comment:
we need the result in the response given that we decode it?

Reply:
we will use it to pass possible errors and/or success situations to the components

p.s. i changed it to `status`